apps/site/lib/site_web/controllers/schedule/finder_api.ex

```diff
@@ -257,11 +257,19 @@ defmodule SiteWeb.ScheduleController.FinderApi do
   defp prepare_journeys_for_json(journeys) do
     journeys
     |> Enum.map(&destruct_journey/1)
+    |> Enum.filter(&journey_has_valid_departure?/1)
     |> Enum.map(&lift_up_route/1)
     |> Enum.map(&set_departure_time/1)
     |> Enum.map(&json_safe_journey/1)
   end
 
+  @spec journey_has_valid_departure?(Journey.t()) :: boolean
+  defp journey_has_valid_departure?(%{departure: departure}) do
+    departure != %{prediction: nil, schedule: nil}
+  end
+
+  defp journey_has_valid_departure?(_), do: true
+
```

Comment:
Pattern matching might work in your favor here!

```suggestion
  defp journey_has_valid_departure?(%{departure: %{prediction: nil, schedule: nil}}), do: false
  defp journey_has_valid_departure?(_), do: true

```

Reply:
Digging around some more I found `PredictedSchedule.empty?` which looks like it could be used here instead. Maybe something like

```elixir
defp journey_has_valid_departure?(%{departure: nil}), do:  true
defp journey_has_valid_departure?(%{departure: departure}), do:  !PredictedSchedule.empty?(departure)
```

Reply:
Wow, that's exactly what we need. Keeping it DRY...